packages/client/src/components/SearchBar/DebouncedText.jsx

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
+
+export function DebouncedText(value, delay) {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [value, delay]);
+
+  return debouncedValue;
+}
```

Comment:
It could be that I am missing the point here, in that case please help me understand 🙏🏽 

This looks like it is neither a component nor a helper function. 
Essentially, it returns the same value that it is taking in as a parameter right away. 

Could you provide and example how this would be implemented to debounce the search?

Reply:
Hi @shpomp this function is being called from another context file which has not been pushed yet, this Debounced function is taking a search text and delay time as parameter and return the same text after little delayed .

Reply:
Hey Madhu! Also chipping in here. Great that you're making a custom hook and moving this piece of complexity to its own file! Much better than adding all this stuff to an existing component

In React hook names always start with `use` so that you can right away see that it's a hook. So in this case, `useDebouncedText` would be a better name :) 

Regarding the PR itself, it would be even better to include this piece of logic together with the other code that makes the debounce work. It's great to have small PR's, but the code should be used :)

And finally, it looks like your code still misses a few things, and that makes sense because debouncing is quite tricky! Personally I also had to implement debounce a few times and I always used this library: https://www.npmjs.com/package/use-debounce. Feel free to use it as well!

And let me know if you have any other questions. Great work!

Reply:
Hi @timnoorlander 
Thanks for review.
We pushed this code with original component so it would be easy to get clear picture of how the whole searching is implemented with debouncing.
In future wherever I need to implement this feature will look into the existing library .
Thanks again .